temporal-sdk/src/main/java/io/temporal/internal/common/SearchAttributesUtil.java

```diff
@@ -0,0 +1,182 @@
+    HistoryLength,
+    RunId,
+    StartTime,
+    StateTransitionCount,
+    TaskQueue,
+    TemporalChangeVersion,
+    WorkflowId,
+    WorkflowType
+  }
+
+  public enum SearchAttributeType {
+    Unspecified,
+    String,
+    Keyword,
+    Int,
+    Double,
+    Bool,
+    Datetime
+  }
+
+  public static Map<String, Payload> serializeToObjectMap(Map<String, Object> searchAttributes) {
+    if (searchAttributes == null) {
+      return null;
+    }
+
+    Map<String, Payload> serializedMap = new HashMap<>();
+    SearchAttributesPayloadConverter dataConverter = new SearchAttributesPayloadConverter();
+    for (Map.Entry<String, Object> item : searchAttributes.entrySet()) {
+      try {
+        serializedMap.put(item.getKey(), dataConverter.toData(item.getValue()).get());
+      } catch (DataConverterException e) {
+        throw new DataConverterException("Cannot serialize key " + item.getKey(), e.getCause());
+      }
+    }
+
+    return serializedMap;
+  }
+
+  public static Map<String, Object> deserializeToObjectMap(SearchAttributes serializedMap) {
+    if (serializedMap == null) {
+      return null;
+    }
+
+    Map<String, Object> deserializedMap = new HashMap<>();
+    for (Entry<String, Payload> attribute : serializedMap.getIndexedFieldsMap().entrySet()) {
+      String key = attribute.getKey();
+      if (!parseSearchAttributes(key, attribute.getValue(), deserializedMap)) {
+        String typeString = attribute.getValue().getMetadataMap().get("type").toStringUtf8();
+        Type javaType = stringToJavaType(typeString);
+        if (javaType == null) {
+          log.error("Error parsing Search Attribute: " + key + " of type " + typeString);
+        } else {
+          deserializedMap.put(
+              key, converter.fromPayload(attribute.getValue(), javaType.getClass(), javaType));
+        }
+      }
+    }
+    return deserializedMap;
+  }
+
+  private static boolean parseSearchAttributes(
+      String key, Payload value, Map<String, Object> deserializedMap) {
+    SearchAttribute searchAttribute;
+    try {
+      searchAttribute = SearchAttribute.valueOf(key);
+    } catch (IllegalArgumentException e) {
+      return false;
+    }
+    switch (searchAttribute) {
+      case BatcherNamespace:
+      case BatcherUser:
+      case BinaryChecksums:
+      case ExecutionStatus:
+      case RunId:
+      case TaskQueue:
+      case TemporalChangeVersion:
+      case WorkflowId:
+      case WorkflowType:
+        deserializedMap.put(key, converter.fromPayload(value, String.class, String.class));
+        break;
+      case CloseTime:
+      case ExecutionTime:
+      case StartTime:
+        deserializedMap.put(
+            key, converter.fromPayload(value, LocalDateTime.class, LocalDateTime.class));
+        break;
+      case ExecutionDuration:
+      case HistoryLength:
+      case StateTransitionCount:
+        deserializedMap.put(key, converter.fromPayload(value, Integer.class, Integer.class));
+        break;
+    }
```

Comment:
default statement?

Reply:
No default statement, because DefaultSearchAttributes.valueOf(key) either resolves into one of the DefaultSearchAttributes or an error occurs.